File: miniapp/pages/myposts/myposts.js
// pages/myposts/myposts.js
const app = getApp();

Page({
  data: {
    currentStatusTab: 'all',
    currentType: '',
    currentTypeText: '全部类型',
    sort: 'time',
    postList: [],
    page: 1,
    limit: 10,
    hasMore: true,
    loading: false,
    loadingMore: false,
    showTypePicker: false,
  },

  onLoad() {
    this.loadData(true);
  },

  loadData(refresh = false) {
    if (refresh) {
      this.setData({ page: 1, postList: [], hasMore: true });
    }
    if (!this.data.hasMore) return;

    this.setData({ loading: refresh, loadingMore: !refresh });

    const params = {
      page: this.data.page,
      limit: this.data.limit,
      sort: this.data.sort
    };
    if (this.data.currentStatusTab !== 'all') {
      params.status = this.data.currentStatusTab;
    }
    if (this.data.currentType) {
      params.type = this.data.currentType;
    }

    wx.request({
      url: `${app.globalData.baseUrl}/api/myposts/list`,
      data: params,
      header: { 'Authorization': 'Bearer ' + wx.getStorageSync('token') },
      success: (res) => {
        if (res.data.code === 0) {
          const list = res.data.data.list || [];
          const total = res.data.data.total || 0;
          const hasMore = this.data.page * this.data.limit < total;
          
          // 处理状态文本汉化（新增pending_confirm）
          list.forEach(item => {
            // 状态文本映射
            item.status_text = {
              'active': '进行中',
              'completed': '已完成',
              'expired': '已过期',
              'off': '已下架',
              'pending': '待审核',
              'rejected': '已驳回',
              'pending_confirm': '待确认' // 新增pending_confirm汉化
            }[item.status] || item.status;
            
            // 类型文本映射（兜底）
            item.type_text = {
              'lost': '寻物',
              'found': '招领',
              'sale': '出售',
              'wanted': '求购'
            }[item.type] || item.type;
          });

          this.setData({
            postList: refresh ? list : this.data.postList.concat(list),
            hasMore: hasMore,
            page: this.data.page + 1
          });
        } else {
          wx.showToast({ title: res.data.message || '加载失败', icon: 'none', duration: 2000 });
        }
      },
      fail: () => {
        wx.showToast({ title: '网络异常', icon: 'none', duration: 2000 });
      },
      complete: () => {
        this.setData({ loading: false, loadingMore: false });
      }
    });
  },

  switchStatusTab(e) {
    const status = e.currentTarget.dataset.status;
    if (status === this.data.currentStatusTab) return;
    this.setData({ currentStatusTab: status });
    this.loadData(true);
  },

  // 新增：设置排序（替换原toggleSort）
  setSort(e) {
    const sort = e.currentTarget.dataset.sort;
    if (sort === this.data.sort) return;
    this.setData({ sort });
    this.loadData(true);
  },

  showTypePicker() {
    this.setData({ showTypePicker: true });
  },

  hideTypePicker() {
    this.setData({ showTypePicker: false });
  },

  selectType(e) {
    const type = e.currentTarget.dataset.type;
    let text = '全部类型';
    if (type) {
      const map = { 'lost': '寻物', 'found': '招领', 'sale': '出售', 'wanted': '求购' };
      text = map[type] || type;
    }
    this.setData({
      currentType: type,
      currentTypeText: text,
      showTypePicker: false
    });
    this.loadData(true);
  },

  loadMore() {
    if (this.data.loadingMore || !this.data.hasMore) return;
    this.loadData();
  },

  goBack() {
    wx.navigateBack({ delta: 1 });
  },

  goToDetail(e) {
    const id = e.currentTarget.dataset.id;
    wx.navigateTo({ url: `/pages/detail/detail?id=${id}` });
  }
});

File: server/app/views/myposts.py
from flask import Blueprint, request, jsonify
from flask_jwt_extended import jwt_required, get_jwt_identity
from app.extensions import db
from app.models.post import Post
from sqlalchemy import case

myposts_bp = Blueprint('myposts', __name__, url_prefix='/api/myposts')

# 状态优先级（用于状态排序）
status_priority = {
    'active': 1,
    'completed': 2,
    'expired': 3,
    'off': 4,
    'pending': 5,
    'rejected': 6
}

@myposts_bp.route('/list', methods=['GET'])
@jwt_required()
def get_myposts():
    user_id = get_jwt_identity()
    status = request.args.get('status')          # 单个状态过滤
    post_type = request.args.get('type')         # 单个类型过滤
    sort = request.args.get('sort', 'time')      # time 或 status
    page = int(request.args.get('page', 1))
    limit = int(request.args.get('limit', 10))
    offset = (page - 1) * limit

    query = Post.query.filter_by(user_id=user_id)

    if status:
        query = query.filter(Post.status == status)
    if post_type:
        query = query.filter(Post.type == post_type)

    if sort == 'status':
        # 按状态优先级排序，相同状态下按发布时间倒序
        query = query.order_by(
            case(
                *[(Post.status == k, v) for k, v in status_priority.items()],
                else_=999
            ),
            Post.created_at.desc()
        )
    else:
        query = query.order_by(Post.created_at.desc())

    total = query.count()
    posts = query.offset(offset).limit(limit).all()

    data = []
    for p in posts:
        first_image = p.images[0] if p.images else None
        data.append({
            'id': p.id,
            'title': p.title,
            'first_image': first_image,
            'type': p.type,
            'type_text': {
                'lost': '寻物',
                'found': '招领',
                'sale': '出售',
                'wanted': '求购'
            }.get(p.type, ''),
            'status': p.status,
            'status_text': {
                'pending': '待审核',
                'active': '进行中',
                'completed': '已完成',
                'expired': '已过期',
                'off': '已下架',
                'rejected': '已驳回'
            }.get(p.status, p.status),
            'created_at': p.created_at.strftime('%Y-%m-%d %H:%M')
        })

    return jsonify({
        'code': 0,
        'data': {
            'list': data,
            'total': total,
            'page': page,
            'limit': limit
        }
    })